examples/empirical_frechet_mean_uncertainty_s3.py

```diff
@@ -0,0 +1,184 @@
+        data = gs.zeros((n, dim+1), dtype=gs.float64)
+        # For sampling on a subsphere, use RandomUniform(dim-1)
+        directions = subsphere.random_uniform(n)
+        # directions = my_random_sample(n, dim-1)
+        for i in range(n):
+            for j in range(dim):
+                data[i,j] = gs.sin(theta) * directions[i,j]
+            data[i,dim] = gs.cos(theta)
+        ## Compute empirical Frechet mean of the n-sample
+        current_mean = sphere.metric.adaptive_gradientdescent_mean(data, n_max_iterations=64, init_points=[north_pole])
+        var.append(sphere.metric.squared_dist(north_pole, current_mean))
+    return (np.mean(var), 2* np.std(var) / np.sqrt( NN ) )
+
+
+def modulation_factor(n, theta, dim, NN=1000):
+    """
+    Compute the modulation factor as the ratio between the variance of
+    the empirical Fréchet mean of n samples drawn from an isotropic
+    distributions on hyper sub-spheres of radius 0 < theta < Pi around
+    the north pole of the sphere of dimension dim.
+    This is repeated NN times to make a stochastic approximation of the
+    expectation and stddev on the modulation factor is returned as second parameter
+    """
+    (var, stdvar) = empirical_var(n, theta, dim, NN=NN)
+    return var * n / theta ** 2, stdvar * n / theta ** 2
+
+
+
+def asymptotic_modulation(dim, theta):
+    gamma = 1.0 / dim + (1.0 - 1.0 / dim) * theta / np.tan(theta)
+    return (1.0 / gamma)**2
+
+
+def plot_modulation_factor(n, dim, NN=1000, Ntheta=20):
+    # modulation factor for n points on the sphere S_dim embedded in R^{dim+1}
+    theta = np.linspace(0.000001, np.pi / 2.0 - 0.000001, Ntheta)
+    measured_modulation_factor = []
+    error = []
+    small_var_modulation_factor = []
+    asymptotic_modulation_factor = []
+    for thetai in theta:
+        (var, stdvar) = modulation_factor(n, thetai, dim, NN=NN)
+        measured_modulation_factor.append(var)
+        error.append(stdvar)
+        print(n, thetai, var, stdvar, '\n')
+        small_var_modulation_factor.append(1.0 + 2.0 / 3.0 * thetai ** 2 * (1.0 - 1.0 / dim) * (1.0 - 1.0 / (n)))
+        asymptotic_modulation_factor.append(asymptotic_modulation(dim, thetai))
+    plt.figure()
+    plt.errorbar(theta, measured_modulation_factor, yerr=error, color='r', label='Measured')  # plotting t,a separately
+    plt.plot(theta, small_var_modulation_factor, 'g', label='Small variance prediction')  # plotting t,a separately
+    plt.plot(theta, asymptotic_modulation_factor, 'grey', label='Asymptotic prediction')  # plotting t,a separately
+    plt.xlabel(r'Standard deviation $\theta$')
+    plt.ylabel(r'Modulation factor $\alpha$')
+    plt.title("Convergence rate modulation factor, sphere dim={1}, n={0}".format(n, dim))
+    plt.legend(loc='best')
+    plt.draw()
+    plt.pause(0.01)
+    ## plt.savefig("Figures/SphVarModulation_N{0}_d{1}.svg".format(n, dim))
+    plt.savefig("Figures/SphVarModulation_N{0}_d{1}.pdf".format(n, dim))
```

Comment:
This line crashes on my laptop (because Figures does not exist), maybe remove it? I'm not sure whether an example should create files.

Reply:
I removed the savefig and kept them for my personal version of this code.